File: Prática/Lista 01 - Java Introdução/Ex13/src/ex13/Ex13.java
package ex13;

import java.util.Random;

/**
 * @author bryan
 */
public class Ex13 {
    
    public static void main(String[] args) {
        
        int[][] matriz = new int[3][3];
        int somaDiagonalPrincipal = 0;
        double mediaDiagonalPrincipal;
        
        GerarMatriz(matriz);

        // --> Somar e obter média da diagonal principal
        for (int i = 0; i < matriz.length; i++) {
            
            somaDiagonalPrincipal += matriz[i][i];
        }
        
        mediaDiagonalPrincipal = (double) somaDiagonalPrincipal / matriz.length;
        
        ExibirMatriz(matriz);
        
        System.out.printf("\nA média da diagonal principal é: %.2f \n\n", mediaDiagonalPrincipal);
        
    }

    // --> Gerar matriz com números inteiros aleatórios de 0 a 99
    public static void GerarMatriz(int[][] matriz) {
        
        Random random = new Random();
        
        for (int linha = 0; linha < matriz.length; linha++) {
            
            for (int coluna = 0; coluna < matriz[0].length; coluna++) {
                matriz[linha][coluna] = random.nextInt(100);
            }
        }
    }
    
    public static void ExibirMatriz(int[][] matriz) {
        
        for (int linha = 0; linha < matriz.length; linha++) {
            
            for (int coluna = 0; coluna < matriz[0].length; coluna++) {
                
                System.out.printf("%2d ", matriz[linha][coluna]);
            }
            
            System.out.println();
        }
    }
}

File: Prática/Lista 04 - C/Ex13/Ex13.c
#include "../Console.c"
#include <stdio.h>
#include <stdlib.h>

int tamanho = 3;

void preecherMatriz(int matriz[][tamanho])
{
    srand(time(NULL));
    for (int linha = 0; linha < tamanho; linha++)
    {
        for (int coluna = 0; coluna < tamanho; coluna++)
        {
            matriz[linha][coluna] = (rand() % 100) + 1;
        }
    }
}

void imprimirMatriz(int matriz[][tamanho])
{
    for (int linha = 0; linha < tamanho; linha++)
    {
        for (int coluna = 0; coluna < tamanho; coluna++)
        {
            printf("%d ", matriz[linha][coluna]);
        }
        printf("\n");
    }
}

float mediaDiagonalPrincipal(int matriz[][tamanho])
{
    int somaDiagonal = 0;

    for (int i = 0; i < tamanho; i++)
    {
        somaDiagonal += matriz[i][i];
    }
    return (float)somaDiagonal/tamanho;
}

int main()
{
    int matriz[tamanho][tamanho];
    preecherMatriz(matriz);
    imprimirMatriz(matriz);
    float mediaDiagonal = mediaDiagonalPrincipal(matriz);
    printf("\nA média da diagonal principal é: %g\n\n", mediaDiagonal);

    return 0;
}

